File: FlutterAppOld/lib/models/table_models/category.dart
class Category {
  int? id;
  String name;

  Category({
    this.id,
    required this.name,
  });

  Map<String, dynamic> toMap() {
    final map = Map<String, dynamic>();
    if (id != null) {
      map['id'] = id;
    }
    map['name'] = name;
    return map;
  }

  factory Category.fromMap(Map<String, dynamic> map) {
    return Category(
      id: map['id'],
      name: map['name'],
    );
  }


}

File: FinanceApp.Shared/Entities/UserTables/Category.cs
using System.ComponentModel.DataAnnotations;

namespace FinanceApp.Shared.Entities.UserTables
{
    public class Category : UserTable
    {
        [Required]
        public string Name { get; set; } = String.Empty;

        public override void CheckInput()
        {
        }
    }
}